ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx

```diff
@@ -0,0 +1,402 @@
+    if (item.isRevision()) {
+      el.toggleClass('is-revision', true)
+      return ++accum
+    }
+    if (item.isDeleted()) {
+      el.toggleClass('is-deleted', true)
+      return ++accum
+    }
+  }, 0)
+
+  el.toggleClass('is-mixed', isMixed > 1)
+
+  const currentWorkspace = store.getCurrentWorkspace()
+  el.toggleClass('in-workspace', Boolean(currentWorkspace))
+  el.toggleClass('is-downloadable', isDownloadable(model))
+  el.toggleClass('is-multiple', model.length > 1)
+  el.toggleClass('is-routed', isRouted())
+  el.toggleClass('is-blacklisted', isBlacklisted(model))
+  el.toggleClass('has-location', hasLocation(model))
+}
+
+const isDownloadable = (model: Model) =>
+  model.find((result: Result) =>
+    result
+      .get('metacard')
+      .get('properties')
+      .get('resource-download-url')
+  )
+const isRouted = () => router && router.toJSON().name === 'openMetacard'
+const isBlacklisted = (model: Model): boolean => {
+  const blacklist = user
+    .get('user')
+    .get('preferences')
+    .get('resultBlacklist')
+  return model.reduce((accum: boolean, result: Result) => {
+    const id = result
+      .get('metacard')
+      .get('properties')
+      .get('id')
+    return blacklist.get(id) || accum
+  }, false)
+}
+
+const isRemoteResourceCached = (model: Model) => {
+  if (!model) return false
+
+  const modelJson = model.toJSON()
+
+  if (!modelJson || modelJson.length <= 0) return false
+
+  return (
+    modelJson[0].isResourceLocal &&
+    modelJson[0].metacard.properties['source-id'] !== sources.localCatalog
+  )
+}
+
+const hasLocation = (model: Model) => getGeoLocations(model).length > 0
+
+const getRegionForSelector = (el: El, selector: any) =>
+  new Marionette.Region({ el: el.find(selector) })
+
+const createAddRemoveRegion = (model: Model) =>
+  PopoutView.createSimpleDropdown({
+    componentToShow: ResultAddView,
+    modelForComponent: model,
+    leftIcon: 'fa fa-plus',
+    rightIcon: 'fa fa-chevron-down',
+    label: 'Add / Remove from List',
+  })
+
+const createResultActionsExportRegion = (model: Model) =>
+  PopoutView.createSimpleDropdown({
+    componentToShow: ExportActionsView,
+    dropdownCompanionBehaviors: {
+      navigation: {},
+    },
+    modelForComponent: model.first(),
+    leftIcon: 'fa fa-external-link',
+    rightIcon: 'fa fa-chevron-down',
+    label: 'Export as',
+  })
+
+const defaultLinks = [
+  {
+    parent: `interaction-add`,
+    dataHelp: `Add the result to a list.`,
+    icon: `fa fa-plus`,
+    linkText: `Add / Remove from List`,
+    actionHandler: handleAdd,
+  },
+  {
+    parent: `interaction-hide`,
+    dataHelp: `Adds to a list
+    of results that will be hidden from future searches.  To clear this list,
+    click the Settings icon, select Hidden, then choose to unhide the record.`,
+    icon: `fa fa-eye-slash`,
+    linkText: `Hide from Future Searches`,
+    actionHandler: handleHide,
+  },
+  {
+    parent: `interaction-show`,
+    dataHelp: `Removes from the
```

Comment:
✏️ The formatting on some of these strings is weird, e.g. this shouldn't have to be on multiple lines.

Reply:
Agreed, but that's the result of `ace format`